src/daft-core/src/series/ops/mod.rs

```diff
@@ -38,21 +38,14 @@ pub mod take;
 mod trigonometry;
 pub mod utf8;
 
-fn match_types_on_series(l: &Series, r: &Series) -> DaftResult<(Series, Series)> {
-    let supertype = try_get_supertype(l.data_type(), r.data_type())?;
-
-    let mut lhs = l.clone();
-
-    let mut rhs = r.clone();
-
-    if !lhs.data_type().eq(&supertype) {
-        lhs = lhs.cast(&supertype)?;
-    }
-    if !rhs.data_type().eq(&supertype) {
-        rhs = rhs.cast(&supertype)?;
-    }
-
-    Ok((lhs, rhs))
+pub fn match_types_on_series(series: Vec<&Series>) -> DaftResult<Vec<Series>> {
```

Comment:
Also now that i think about it the function name for this is a little misleading, what it actually does is casting series to their supertype not rlly type matching, maybe rename to 'cast_series_to_supertype` ?